File: 2021/June/Day 24/powerOfThree.cpp
// Source : https://leetcode.com/problems/check-if-number-is-a-sum-of-powers-of-three/submissions/
class Solution {
public:
    bool checkPowersOfThree(int n) {

        while(n)
        {
            if(n%3==2){return false;}
            n/=3;
        }
        return true;
    }
};

File: 2024/Jul/powerOfThree.go
// Source : https://leetcode.com/problems/power-of-three/
func isPowerOfThree(n int) bool {
	for n >= 1 {
		if n == 1 {
			return true
		} else if n == 0 {
			return false
		}
		if n%3 != 0 {
			return false
		}
		n /= 3
	}
	return false
}